feature-based-molecular-networking/tools/feature-based-molecular-networking/scripts/progenesis_formatter.py

```diff
@@ -94,50 +104,43 @@ def convert_to_feature_csv(input_filename, output_filename):
         newdf.rename(columns={'CCS (angstrom^2)':'CCS'}, inplace=True)
     if 'dCCS (angstrom^2)' in newdf.columns:
         newdf.rename(columns={'dCCS (angstrom^2)':'dCCS'}, inplace=True)
-        
-    # Drop useless duplicated columns
+
+    ## Drop useless duplicated columns
     if 'm/z' in newdf.columns:
         newdf = newdf.drop(['m/z'], axis=1)
     if 'Retention time (min)' in newdf.columns:
         newdf = newdf.drop(['Retention time (min)'], axis=1)
-        
+
     #Round values for columns
     newdf['Neutral mass'] = newdf['Neutral mass'].apply(lambda x: round(x, 4))
     newdf['Chromatographic peak width (min)'] = newdf['Chromatographic peak width (min)'].apply(lambda x: round(x, 3))
 
-    # Create the output table
-    newdf_out = pd.concat([output_df_prepared,newdf],axis=1)
-
-    # Remove duplicated columns
-    if 'Compound' in newdf_out.columns: 
-        newdf_out = newdf_out.drop(newdf_out.columns[0], axis=1)
-    if 'Isotope Distribution' in newdf_out.columns: 
-        newdf_out = newdf_out.drop(newdf_out.columns[0], axis=1)
-     
-    #Drop empty column   
-    newdf_out = newdf_out.dropna(how='all', axis=1)
-        
-    newdf_out.to_csv(output_filename, sep=",", index=False)
-    
+    #Write out the table
+    newdf.to_csv(str(output_filename), sep=",", index=False)
+
     return compound_to_scan_mapping
 
 #Converts MSP to MGF
 def convert_mgf(input_msp, output_mgf, compound_to_scan_mapping):
     output_filename = open(output_mgf, "w")
     read_name = False
-    
+
     scan = -1
     precursor_mz = -1
     charge = -1
     peaks = []
 
-    # This is a stop gap solution to make sure we don't have repetitions in the MGF file. 
+    # This is a stop gap solution to make sure we don't have repetitions in the MGF file.
     # We only take one MS2 arbitrarily selected and output into the MGF file
     observed_compound_names = set()
 
     for line in open(input_msp):
```

Comment:
What was qualitatively changed here?

Reply:
It is the same imperfect solution (only one MS2 per compound), but we add more qualitative information that are important for the network (the principal one is the charge, but also adding the adduct, retention time, and file.

Reply:
Can you point me to an example output file in the test folder?

Reply:
It works for all test files (see test_converters.py)

The test files are in:
feature-based-molecular-networking/tools/feature-based-molecular-networking/test/Progenesis/ 
Including 
161118_pos_IMS.txt -> 161118_pos_IMS_out.csv
161118_pos_IMS.msp -> 161118_pos_IMS.mgf